### PR TITLE
chore: add native image sample for google-cloud-secretmanager

### PR DESCRIPTION
**Pull request by mpeddada1**

### CODE REVIEW

**Comment by mpeddada1** [comment]
The `generation diff / root-pom` check failure is expected as this sample module hasn't been included in https://github.com/googleapis/sdk-platform-java/blob/main/library_generation/templates/root-pom.xml.j2 yet 

(thanks for the context @JoeWang1127)

**Comment by mpeddada1** [comment]
The presubmit is detecting changes in the sample:

Screenshot 2024-05-13 at 7 07 11 PM

**Comment by suztomo** [comment]
@JoeWang1127 How does the hermetic build script handles the sample directory?

**Comment by JoeWang1127** [comment]
> @JoeWang1127 How does the hermetic build script handles the sample directory?

Hermetic build won't affect `java-samples` because it's not listed in `generation_config.yaml`

File: java-samples/native-image-sample/pom.xml
Comment (burkedavison) [inline java-samples/native-image-sample/pom.xml:128]: Should we be getting any of these versions from an upstream source to make sure they stay in sync with our test infrastructure?

File: pom.xml
Comment (burkedavison) [reply pom.xml:157]: The intent was to use the [same solution used in spring-cloud-gcp](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-samples/pom.xml#L265-L270), and avoid the need for special CI logic. 

This is what prevents the sample from being deployed:
https://github.com/googleapis/google-cloud-java/pull/10805/files#diff-27d2eee66ae8d7e40703e2670c7107abc6667008967085105fab6f87624aeff9R41-R47

File: java-samples/native-image-sample/pom.xml
Comment (mpeddada1) [reply java-samples/native-image-sample/pom.xml:129]: That's a good point. Proposed a PR in java-shared-config to manage the dependency versions in Maven properties: https://github.com/googleapis/java-shared-config/pull/824

File: pom.xml
Comment (mpeddada1) [reply pom.xml:157]: > The intent was to use the [same solution used in spring-cloud-gcp](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/spring-cloud-gcp-samples/pom.xml#L265-L270), and avoid the need for special CI logic.

+1 to add a little bit more context to this - Without this change or special customizations to the script, we run into the following error:
```
18:33:33:151 [WARNING] 
18:33:33:151 [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
18:33:33:152 [WARNING] 
18:33:33:152 [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
18:33:33:152 [WARNING] 
18:33:33:221 [ERROR] [ERROR] Could not find the selected project in the reactor: java-samples/native-image-sample @ 
18:33:33:222 [ERROR] Could not find the selected project in the reactor: java-samples/native-image-sample -> [Help 1]
18:33:33:222 [ERROR] 
18:33:33:223 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
18:33:33:223 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

Comment (burkedavison) [reply pom.xml:157]: Yeah - removing the module from the parent means needing to `cd` into the sample directory and run `mvn` commands from there. So it'd be a dedicated set of scripts to perform that testing.

**Comment by burkedavison** [comment]
This LGTM, but I'd like to have @suztomo 's approval before merge.

File: java-samples/pom.xml
Comment (suztomo) [inline java-samples/pom.xml:8]: Can you make the version as "0.1.0-SNAPSHOT" without Release Please annotations? With that, we ensure that the module won't be published to Maven Central.

File: pom.xml
Comment (suztomo) [reply pom.xml:157]: I see. To add precaution, would you set the versions of the sample modules as "1.0-SNAPSHOT" without the Release Please annotation? That ensures the modules won't be published to Maven Central accidentally.

File: java-samples/native-image-sample/pom.xml
Comment (suztomo) [inline java-samples/native-image-sample/pom.xml:9]: Can you make the version as "0.1.0-SNAPSHOT" without Release Please annotations?

**Review by suztomo — CHANGES_REQUESTED** [review]

File: pom.xml
Comment (mpeddada1) [reply pom.xml:157]: Done, thank you.

File: java-samples/native-image-sample/pom.xml
Comment (mpeddada1) [reply java-samples/native-image-sample/pom.xml]: Done. Modified this files and java-samples/pom.xml to use `0.1.0-SNAPSHOT`

File: java-samples/pom.xml
Comment (mpeddada1) [reply java-samples/pom.xml]: Done, that's a good idea.

**Review by suztomo — APPROVED** [review]
Approving this.Please confirm the test job log before merging this.

**Comment by mpeddada1** [comment]
Thank you! Confirmed by running the stage job with this PR's commit that we aren't staging any of the sample artifacts.

**Pull request merged**
